tests/mve/gtest_scene.cc

```diff
@@ -0,0 +1,715 @@
+void make_dirty (mve::View::Ptr view);
+void make_a_clean_view_dirty (mve::Scene::Ptr scene);
+mve::Scene::ViewList load_views_from (const std::string &scene_path);
+mve::Bundle::Ptr load_bundle_from (const std::string &scene_path);
+mve::Bundle::Ptr make_bundle (const std::size_t camera_count);
+bool views_match (mve::Scene::ViewList& lhs, mve::Scene::ViewList& rhs);
+bool bundle_cameras_match (mve::Bundle::Ptr lhs, mve::Bundle::ConstPtr rhs);
+
+} // anonymous namespace
+
+//== Test the initial state of a created scene =================================
+
+TEST(SceneTest, ACreatedSceneIsInitialyClean)
+{
+    OnScopeExit clean_up;
+
+    std::string scene_path = create_scene_on_disk(0, nullptr, clean_up);
+    mve::Scene::Ptr scene = mve::Scene::create(scene_path);
+    ASSERT_TRUE(scene != nullptr);
+    EXPECT_FALSE(scene->is_dirty());
+}
+
+TEST(SceneTest, TheInitialPathOfACreatedSceneIsThePathItWasCreatedWith)
+{
+    OnScopeExit clean_up;
+
+    std::string scene_path = create_scene_on_disk(0, make_bundle(0), clean_up);
+    mve::Scene::Ptr scene = mve::Scene::create(scene_path);
+    EXPECT_EQ(scene_path, scene->get_path());
+}
+
+TEST(SceneTest, TheInitialViewsOfACreatedSceneMatchWithThatSceneOnDisk)
+{
+    using ViewList = mve::Scene::ViewList;
+    OnScopeExit clean_up;
+
+    mve::Scene::Ptr scene_without_views = [&clean_up]()
+    {
+        std::string path = create_scene_on_disk(0, make_bundle(5), clean_up);
+        return mve::Scene::create(path);
+    }();
+    EXPECT_EQ(scene_without_views->get_views().size(), std::size_t(0));
+
+    mve::Scene::Ptr scene_with_views = [&clean_up]()
+    {
+        std::string path = create_scene_on_disk(73, make_bundle(23), clean_up);
+        return mve::Scene::create(path);
+    }();
+    ViewList views_on_disk = load_views_from(scene_with_views->get_path());
+    EXPECT_TRUE(views_match(views_on_disk, scene_with_views->get_views()));
+}
+
+TEST(SceneTest, TheInitialBundleOfACreatedSceneMatchesWithThatSceneOnDisk)
+{
+    OnScopeExit clean_up;
+
+    mve::Scene::Ptr scene_with_empty_bundle = [&clean_up]()
+    {
+        std::string path = create_scene_on_disk(0, make_bundle(0), clean_up);
+        return mve::Scene::create(path);
+    }();
+    EXPECT_TRUE(bundle_cameras_match(
+        load_bundle_from(scene_with_empty_bundle->get_path()),
+        scene_with_empty_bundle->get_bundle()));
+
+    mve::Scene::Ptr scene_with_non_empty_bundle = [&clean_up]()
+    {
+        std::string path = create_scene_on_disk(3, make_bundle(23), clean_up);
+        return mve::Scene::create(path);
+    }();
+    EXPECT_TRUE(bundle_cameras_match(
+        load_bundle_from(scene_with_non_empty_bundle->get_path()),
+        scene_with_non_empty_bundle->get_bundle()));
+}
+
+//== Creating a scene with missing files or directories ========================
+
+TEST(SceneTest, CreateSceneThrowsAnExceptionIfTheDirectoryDoesNotExist)
+{
+    std::string not_a_directory = tmpnam(nullptr);
```

Comment:
Not sure what you mean here.
Most test cases use create_scene_on_disk to make a populated scene directory to test with, but this particular test needs a path to a directory that does not exist.